training/src/main/scala/com/airbnb/aerosolve/training/NDTree.scala

```diff
@@ -186,13 +206,13 @@ object NDTree {
       axisIndex: Int): Double = {
     // TODO (christhetree): use median of medians algorithm (quick select) for O(n)
     // performance instead of O(nln(n)) performance
```

Comment:
breeze has quick select implementation. we can make this change if this is a hot spot in NDTree.